### PR TITLE
Limit memory use during 32bit build

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: sysimage.mk [hunk 1]
```diff
@@ -82,7 +82,7 @@ $$(build_private_libdir)/sys$1-o.a $$(build_private_libdir)/sys$1-bc.a : $$(buil
 		 JULIA_PROJECT= \
 		 JULIA_DEPOT_PATH=':' \
 		 JULIA_NUM_THREADS=1 \
-			$$(call spawn, $3) $2 -C "$$(JULIA_CPU_TARGET)" --output-$$* $$(call cygpath_w,$$@).tmp $$(JULIA_SYSIMG_BUILD_FLAGS) \
+			$$(call spawn, $3) $2 -C "$$(JULIA_CPU_TARGET)" "$$(HEAPLIM)" --output-$$* $$(call cygpath_w,$$@).tmp $$(JULIA_SYSIMG_BUILD_FLAGS) \
```
Comment (vchuravy) [inline sysimage.mk:85]:
```suggestion
			$$(call spawn, $3) $2 -C "$$(JULIA_CPU_TARGET)" $$(HEAPLIM) --output-$$* $$(call cygpath_w,$$@).tmp $$(JULIA_SYSIMG_BUILD_FLAGS) \
```

File: contrib/generate_precompile.jl [hunk 2]
```diff
@@ -465,6 +465,7 @@ generate_precompile_statements() = try # Make sure `ansi_enablecursor` is printe
     print("Total ─────── "); Base.time_print(stdout, tot_time); println()
 finally
     fancyprint && print(ansi_enablecursor)
+    GC.gc(true) # reduce memory footprint
```
Comment (d-netto) [inline contrib/generate_precompile.jl:468]: Perhaps replace this with `GC.gc(true); GC.gc(false)`?

`GC.gc(true)` just triggers a full sweep, which means that old garbage will be collected in the next GC cycle.

We might want to insert a `GC.gc(false)` to expedite this subsequent collection of garbage.

Comment (vchuravy) [inline contrib/generate_precompile.jl:468]:
```suggestion
    GC.gc(true); GC.gc(false); # reduce memory footprint
```

Comment (gbaraldi) [reply contrib/generate_precompile.jl]: Don't we do a recollect with the manual GC?

**Review by staticfloat — APPROVED** [review]

**Pull request merged**